src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -0,0 +1,434 @@
+
+            public IWebHostBuilder ConfigureServices(Action<WebHostBuilderContext, IServiceCollection> configureServices)
+            {
+                configureServices(new WebHostBuilderContext
+                {
+                    Configuration = _configuration,
+                    HostingEnvironment = _environment
+                },
+                _services);
+                return this;
+            }
+
+            public IWebHostBuilder ConfigureServices(Action<IServiceCollection> configureServices)
+            {
+                return ConfigureServices((WebHostBuilderContext context, IServiceCollection services) => configureServices(services));
+            }
+
+            public string? GetSetting(string key)
+            {
+                _settings.TryGetValue(key, out var value);
+                return value;
+            }
+
+            public IWebHostBuilder UseSetting(string key, string? value)
+            {
+                _settings[key] = value;
+                _operations += b => b.UseSetting(key, value);
+
+                // All preoperties on IWebHostEnvironment are non-nullable.
+                if (value is null)
+                {
+                    return this;
+                }
+
+                if (key == WebHostDefaults.ApplicationKey)
+                {
+                    _environment.ApplicationName = value;
+                }
+                else if (key == WebHostDefaults.ContentRootKey)
+                {
+                    _environment.ContentRootPath = value;
+                    _environment.ResolveFileProviders(_configuration);
+
+                    _configuration.ChangeBasePath(value);
+                }
+                else if (key == WebHostDefaults.EnvironmentKey)
+                {
+                    _environment.EnvironmentName = value;
+                }
+                else if (key == WebHostDefaults.WebRootKey)
+                {
+                    _environment.WebRootPath = value;
+                    _environment.ResolveFileProviders(_configuration);
+                }
+
+                return this;
+            }
+
+            public void ExecuteActions(IWebHostBuilder webHostBuilder)
+            {
+                _operations?.Invoke(webHostBuilder);
+            }
+        }
+
+        private class LoggingBuilder : ILoggingBuilder
+        {
+            public LoggingBuilder(IServiceCollection services)
+            {
+                Services = services;
+            }
+
+            public IServiceCollection Services { get; }
+        }
+
+        private class WebHostEnvironment : IWebHostEnvironment
+        {
+            private static readonly NullFileProvider NullFileProvider = new();
+
+            public WebHostEnvironment(Assembly? callingAssembly)
+            {
+                ContentRootPath = Directory.GetCurrentDirectory();
+
+                ApplicationName = (callingAssembly ?? Assembly.GetEntryAssembly())?.GetName()?.Name ?? "NotFound";
```

Comment:
NotFound? I'm not sure what we do normally here but we should check.

Reply:
It looks like [we use string.Empty](https://github.com/dotnet/aspnetcore/blob/a4a3668c455fe87f731803e4d0b3abfbc4f22198/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L125).

There being no name is an edge case. I'm not even sure how to force it. Do we really prefer `""` over `"NotFound"`?

Reply:
I would prefer empty yes. I think when you do single file maybe?

Reply:
I tried single file publish on Windows to see. No difference. I bet you could force it with IL Emit or something, but that's too much work. I switched it to `string.Empty` for consistency.